rfcs/2021-09-01-8547-accept-metrics-in-datadog-agent-source.md

```diff
@@ -0,0 +1,245 @@
+# RFC 8547 - 2021-09-01 - Accept metrics in `datadog_agent` source
+
+Currently the `datadog_agent` [source](https://vector.dev/docs/reference/configuration/sources/datadog_agent/) only
+supports logs. This RFC suggests to extend Vector to support receiving metrics from Datadog agents and ingest those as
+metrics from a Vector perspective so they can be benefit from Vector capabilities.
+
+  * [Context](#context)
+  * [Cross cutting concerns](#cross-cutting-concerns)
+  * [Scope](#scope)
+    + [In scope](#in-scope)
+    + [Out of scope](#out-of-scope)
+  * [Pain](#pain)
+  * [Proposal](#proposal)
+    + [User Experience](#user-experience)
+    + [Implementation](#implementation)
+  * [Rationale](#rationale)
+  * [Drawbacks](#drawbacks)
+  * [Prior Art](#prior-art)
+  * [Alternatives](#alternatives)
+    + [For transport between Agents and Vector](#for-transport-between-agents-and-vector)
+    + [Flattening sketches](#flattening-sketches)
+    + [For Request routing](#for-request-routing)
+  * [Outstanding Questions](#outstanding-questions)
+  * [Plan Of Attack](#plan-of-attack)
+  * [Future Improvements](#future-improvements)
+
+## Context
+
+- Vector is foreseen as a Datadog Agents aggregator, thus receiving metrics from Datadog Agents is a logical development
+- Vector has support to send metrics to Datadog, thus receiving metrics from Agent is a consistent feature to add
+
+## Cross cutting concerns
+
+Some known issues are connected to the work described here: [#7283], [#8493] & [#8626]. This mostly concerns the ability
+to store/manipulate distribution using [sketches], send those to Datadog using the DDSketch representation. Other
+metrics sinks would possibly benefit from having distribution stored internally with sketches as this would provide
+better aggregation and accuracy.
+
+[#7283]: https://github.com/timberio/vector/issues/7283
+[#8493]: https://github.com/timberio/vector/issues/8493
+[#8626]: https://github.com/timberio/vector/issues/8626
+[sketches]: http://www.vldb.org/pvldb/vol12/p2195-masson.pdf
+
+## Scope
+
+### In scope
+
+- Implement a Datadog metrics endpoint in Vector, it will match the [metrics intake
+  API](https://docs.datadoghq.com/api/latest/metrics/) with additional route that the Agent uses
+- Include support for sketches that uses protobuf.
+- Ensure all Datadog metrics type are mapped to internal Vector metric type and that there is no loss of accuracy in a
+  pass through configuration.
+
+### Out of scope
+
+- Anything not related to metrics
+  - Processing API validation requests
+  - Processing other kind of payloads: traces, event, etc.
+- Shipping sketches to Datadog in the `datadog_metrics` sinks, it is required reach a fully functional situation but
+  this is not the goal of this RFC that focus on receiving metrics from Datadog Agents.
+
+## Pain
+
+- Users cannot aggregate metrics from Datadog agents
+
+## Proposal
+
+### User Experience
```

Comment:
Doesn't this section need to be updated with the new direction?

Reply:
Indeed, I just pushed an update to reflect the new direction.